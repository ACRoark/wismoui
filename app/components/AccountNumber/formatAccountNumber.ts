const formatAccountNumber = (value: string): string => {
  const regex = new RegExp('^[0-9]{12}$', 'g');

  if (!regex.test(value)) {
    throw new Error('Account numbers must be 12 numeric digits.');
  }

  const formattedValue = `${value.substr(0, 4)} ${value.substr(4, 3)} ${value.substr(7, 4)} ${value.substr(11, 1)}`;

  return formattedValue;
};

export default formatAccountNumber;
