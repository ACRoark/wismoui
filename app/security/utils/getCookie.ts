const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
