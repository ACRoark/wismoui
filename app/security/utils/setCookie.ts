interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
