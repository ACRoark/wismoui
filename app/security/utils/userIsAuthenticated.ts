import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
