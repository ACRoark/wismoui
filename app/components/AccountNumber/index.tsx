import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
