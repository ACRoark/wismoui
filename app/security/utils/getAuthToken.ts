import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
