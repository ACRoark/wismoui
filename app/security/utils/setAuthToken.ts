import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
