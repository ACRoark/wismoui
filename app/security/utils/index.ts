import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
