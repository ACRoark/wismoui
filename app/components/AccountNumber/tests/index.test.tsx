import { render } from 'enzyme';
import React from 'react';

import { createSnapshot } from 'testing/utils';

import AccountNumber from '..';

describe('<AccountNumber />', (): void => {
  it('should not log errors in console when rendered', (): void => {
    const spy = jest.spyOn(global.console, 'error');

    render(<AccountNumber value="112233445566" />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('should render correctly when passed a valid account number value', (): void => {
    const tree = createSnapshot(<AccountNumber value="112233445566" />);

    expect(tree).toMatchSnapshot();
  });
});
