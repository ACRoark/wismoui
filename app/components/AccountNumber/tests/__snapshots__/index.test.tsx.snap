// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountNumber /> should render correctly when passed a valid account number value 1`] = `
<span>
  1122 334 4556 6
</span>
`;
